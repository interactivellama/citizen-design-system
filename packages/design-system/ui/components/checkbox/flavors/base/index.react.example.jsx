// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import classNames from 'classnames';
import _ from 'lodash';

///////////////////////////////////////////
// Partial(s)
///////////////////////////////////////////

let Demo = props =>
  <div className="demo-only" {...props}>
    {props.children}
  </div>;

let Fieldset = props =>
  <fieldset className={classNames('slds-form-element', props.className)}>
    {props.children}
  </fieldset>;

let Legend = props =>
  <legend className="slds-form-element__legend slds-form-element__label">{props.children}</legend>;

let FormElement = props =>
  <div className={classNames('slds-form-element', props.className)}>
    {props.children}
  </div>;

let FormElementLabel = props =>
  <label className="slds-form-element__label" htmlFor={props.id}>Checkbox Label</label>;

let FormElementControl = props =>
  <div className={classNames('slds-form-element__control', props.className)}>
    {props.children}
  </div>;

export let Checkbox = props => {
  const uniqueId = _.uniqueId('checkbox-');

  return (
    <span className="slds-checkbox">
      {props.children}
      <input type="checkbox" name={ props.name || 'options' } id={ props.id ? props.id : uniqueId } disabled={props.disabled} defaultChecked={props.checked} aria-describedby={props.errorId} />
      <label className={classNames('slds-checkbox__label', props.className)} htmlFor={ props.id ? props.id : uniqueId }>
        <span className="slds-checkbox--faux"></span>
        <span className={classNames('slds-form-element__label', props.hideLabel ? 'slds-assistive-text' : null)}>{props.label}</span>
      </label>
    </span>
  );
};

//////////////////////////////////////////////
// State Constructor(s)
//////////////////////////////////////////////

let Default = props =>
  <Demo>
    <FormElement>
      <FormElementControl>
        <Checkbox label="Checkbox Label" checked />
      </FormElementControl>
    </FormElement>
  </Demo>;

let Indeterminate = props =>
  <Demo>
    <FormElement>
      <FormElementControl>
        <Checkbox id="checkbox-indeterminate-01" label="Indeterminate Checkbox Label" />
      </FormElementControl>
    </FormElement>
  </Demo>;

let Required = props =>
  <Demo>
    <FormElement>
      <FormElementControl>
        <Checkbox label="Checkbox Label"><abbr className="slds-required" title="required">*</abbr></Checkbox>
      </FormElementControl>
    </FormElement>
  </Demo>;

let ErrorState = props =>
  <Demo>
    <FormElement className="slds-has-error">
      <FormElementControl>
        <Checkbox label="Checkbox Label" errorId="error_01"><abbr className="slds-required" title="required">*</abbr></Checkbox>
      </FormElementControl>
      <div id="error_01" className="slds-form-element__help">This field is required</div>
    </FormElement>
  </Demo>;

let Disabled = props =>
  <Demo>
    <FormElement>
      <FormElementControl>
        <Checkbox label="Checkbox Label" disabled />
      </FormElementControl>
    </FormElement>
  </Demo>;

let Group = props =>
  <Demo>
    <Fieldset>
      <Legend>Checkbox Group Label</Legend>
      <FormElementControl>
        <Checkbox label="Checkbox Label" checked />
        <Checkbox label="Checkbox Label" />
      </FormElementControl>
    </Fieldset>
  </Demo>;

let GroupRequired = props =>
  <Demo>
    <Fieldset>
      <Legend><abbr className="slds-required" title="required">*</abbr> Checkbox Group Label</Legend>
      <FormElementControl>
        <Checkbox label="Checkbox Label" checked />
        <Checkbox label="Checkbox Label" />
      </FormElementControl>
    </Fieldset>
  </Demo>;

let GroupError = props =>
  <Demo>
    <Fieldset className="slds-has-error">
      <Legend><abbr className="slds-required" title="required">*</abbr> Checkbox Group Label</Legend>
      <FormElementControl>
        <Checkbox errorId="error_01" label="Checkbox Label" checked />
        <Checkbox errorId="error_01" label="Checkbox Label" />
      </FormElementControl>
      <div id="error_01" className="slds-form-element__help">This field is required</div>
    </Fieldset>
  </Demo>;

let GroupDisabled = props =>
  <Demo>
    <Fieldset>
      <Legend>Checkbox Group Label</Legend>
      <FormElementControl>
        <Checkbox label="Checkbox Label" disabled></Checkbox>
        <Checkbox label="Checkbox Label" disabled></Checkbox>
      </FormElementControl>
    </Fieldset>
  </Demo>;

//////////////////////////////////////////////
// Export
//////////////////////////////////////////////

export let states = [
  {
    id: 'checkbox',
    label: 'Default',
    element: <Default />
  },
  {
    id: 'checkbox-indeterminate',
    label: 'Indeterminate',
    element: <Indeterminate />,
    script: `
      var checkbox = document.getElementById('checkbox-indeterminate-01')
      checkbox.indeterminate = true
    `
  },
  {
    id: 'checkbox-required',
    label: 'Required',
    element: <Required />
  },
  {
    id: 'checkbox-error',
    label: 'Error',
    element: <ErrorState />
  },
  {
    id: 'checkbox-disabled',
    label: 'Disabled',
    element: <Disabled />
  },
  {
    id: 'checkbox-group',
    label: 'Group',
    element: <Group />
  },
  {
    id: 'checkbox-group-required',
    label: 'Group Required',
    element: <GroupRequired />
  },
  {
    id: 'checkbox-group-error',
    label: 'Group with error',
    element: <GroupError />
  },
  {
    id: 'checkbox-group-disabled',
    label: 'Group Disabled',
    element: <GroupDisabled />
  }
];
