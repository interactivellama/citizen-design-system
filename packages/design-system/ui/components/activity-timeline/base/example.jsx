// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { ButtonIcon } from 'ui/components/button-icons/base/example';
import { ActivityTimelineItem as Event } from 'ui/components/activity-timeline/event/example';
import { ActivityTimelineItem as Call } from 'ui/components/activity-timeline/call/example';
import { ActivityTimelineItem as Email } from 'ui/components/activity-timeline/email/example';
import { ActivityTimelineItem as Task } from 'ui/components/activity-timeline/task/example';
import SvgIcon from 'app_modules/ui/svg-icon';
import classNames from 'classnames';

export let TimelineIcon = props =>
  <div
    className={classNames(
      'slds-icon_container',
      props.symbol == 'log_a_call' ?
        'slds-icon-standard-log-a-call'
      : 'slds-icon-standard-' + props.symbol
    )}
    title={props.symbol == 'log_a_call' ? 'call' : props.symbol }
  >
    <SvgIcon
      className="slds-icon slds-icon--small"
      sprite="standard"
      symbol={ props.symbol }
    />
  </div>;

export let TimelineActions = props =>
  <div className="slds-timeline__actions">
    <p className="slds-timeline__date">{ props.date || 'Feb 24' }</p>
    <ButtonIcon
      className="slds-button--icon-border-filled slds-button--icon-x-small"
      symbol="down"
      assistiveText={ props.assistiveText || 'More Options' }
      aria-haspopup="true"
      title={ props.title || 'More Options' }
    />
  </div>;

export default (
  <ul className="slds-timeline">
    <li>
      <Task />
    </li>
    <li>
      <Event />
    </li>
    <li>
      <Call />
    </li>
    <li>
      <Email />
    </li>
  </ul>
);
