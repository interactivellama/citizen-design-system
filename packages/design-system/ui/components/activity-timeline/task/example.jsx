// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import MediaObject from 'ui/utilities/media-objects/index.react';
import { Checkbox } from 'ui/components/checkbox/base/example';
import { TimelineIcon, TimelineActions } from 'ui/components/activity-timeline/base/example';

export let ActivityTimelineItem = props =>
  <div className="slds-timeline__item">
    <span className="slds-assistive-text">Task</span>
    <MediaObject
      figureRight={
        <TimelineActions
          title="More Options for Task, Review proposals"
          assistiveText="More Options for Task, Review proposals"
        />
      }
    >
      <MediaObject
        className="slds-timeline__media slds-timeline__media--task"
        figureLeft={ <TimelineIcon symbol="task" /> }
        figureLeftClassName="slds-timeline__icon"
      >
        <MediaObject
          title="Review proposals for EBC deck with larger team and have marketing review this"
          figureLeft={ <Checkbox hideLabel label="mark-complete" /> }
        >
          <h3 className="slds-truncate" title="Review proposals for EBC deck with larger team and have marketing review this">
            <a href="javascript:void(0);">Review proposals for EBC deck with larger team and have marketing review this</a>
          </h3>
          <ul className="slds-list--horizontal slds-wrap">
            <li className="slds-m-right--large">
              <span className="slds-text-title">Contact:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Lei Chan</a></span>
            </li>
            <li className="slds-m-right--large">
              <span className="slds-text-title">Assigned to:</span>
              <span className="slds-text-body--small"><a href="javascript:void(0);">Betty Mason</a></span>
            </li>
          </ul>
        </MediaObject>
      </MediaObject>
    </MediaObject>
  </div>;

export default (
  <ActivityTimelineItem />
);
