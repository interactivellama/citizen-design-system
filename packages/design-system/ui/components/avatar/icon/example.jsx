// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import SvgIcon from 'app_modules/ui/svg-icon';
import { StandardIcon } from 'ui/components/icons/standard/example';
import { Avatar } from 'ui/components/avatar/base/example';
import classNames from 'classnames';


//////////////////////////////////////////////
// Export
//////////////////////////////////////////////

export default (
  <Avatar>
    <StandardIcon />
  </Avatar>
);

export let states = [
  {
    id: 'avatar-square-icon',
    label: 'Square Icon Fallback',
    element:
      <Avatar className="slds-avatar--medium">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'avatar-circle-icon',
    label: 'Circle Icon Fallback',
    element:
      <Avatar className="slds-avatar--circle slds-avatar--medium">
        <StandardIcon symbol="user" />
      </Avatar>
  },
  {
    id: 'size-x-small',
    label: 'Size - X Small',
    element:
      <Avatar className="slds-avatar--x-small">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-small',
    label: 'Size - Small',
    element:
      <Avatar className="slds-avatar--small">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-medium',
    label: 'Size - Medium',
    element:
      <Avatar className="slds-avatar--medium">
        <StandardIcon />
      </Avatar>
  },
  {
    id: 'size-large',
    label: 'Size - Large',
    element:
      <Avatar className="slds-avatar--large">
        <StandardIcon />
      </Avatar>
  }
];
