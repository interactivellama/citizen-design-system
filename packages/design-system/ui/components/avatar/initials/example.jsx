// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { Avatar } from '../base/example';
import { AvatarInitials } from '../';

export default (
  <Avatar>
    <AvatarInitials
      iconClassName="slds-icon-standard-account"
      title="company name"
      initials="Ac"
    />
  </Avatar>
);

export let examples = [
  {
    id: 'linked-avatar-initials',
    label: 'Wrapped in a link',
    element: (
      <a href="javascript:void(0);">
        <Avatar>
          <AvatarInitials
            iconClassName="slds-icon-standard-account"
            title="company name"
            initials="Ac"
          />
        </Avatar>
      </a>
    )
  }
];
