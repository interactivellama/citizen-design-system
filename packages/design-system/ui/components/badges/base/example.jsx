// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { UtilityIcon } from '../../icons/base/example';
import { Badge, InverseBadge, LightestBadge } from '../';

export default <Badge>Badge Label</Badge>;

export let examples = [
  {
    id: 'inverse',
    label: 'Inverse',
    element: <InverseBadge>Badge Label</InverseBadge>
  },
  {
    id: 'light-with-icon',
    label: 'Light with Icon',
    element: (
      <LightestBadge>
        <UtilityIcon
          containerClassName="slds-m-right_xx-small"
          className="slds-icon_xx-small slds-icon-text-default"
          assistiveText={false}
          symbol="moneybag"
        />
        423 Credits Available
      </LightestBadge>
    )
  },
  {
    id: 'badge-with-child',
    label: 'With Child Element',
    element: (
      <Badge>
        Component In: <strong>Lightning</strong>
      </Badge>
    )
  }
];
