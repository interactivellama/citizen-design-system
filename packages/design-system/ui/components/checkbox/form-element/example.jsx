// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import { CheckboxStandalone } from '..';
import { FormElement } from '../../form-element/';
import { UtilityIcon } from '../../icons/base/example';

export default (
  <FormElement labelContent="Form Element Label" inputId="checkbox-id-01">
    <CheckboxStandalone id="checkbox-id-01" />
  </FormElement>
);

export let states = [
  {
    id: 'checked',
    label: 'Checked',
    element: (
      <FormElement labelContent="Form Element Label" inputId="checkbox-id-01">
        <CheckboxStandalone id="checkbox-id-01" isChecked />
      </FormElement>
    )
  },
  {
    id: 'disabled',
    label: 'Disabled',
    element: (
      <FormElement labelContent="Form Element Label" inputId="checkbox-id-01">
        <CheckboxStandalone id="checkbox-id-01" isDisabled />
      </FormElement>
    )
  },
  {
    id: 'checked-disabled',
    label: 'Checked and Disabled',
    element: (
      <FormElement labelContent="Form Element Label" inputId="checkbox-id-01">
        <CheckboxStandalone id="checkbox-id-01" isChecked isDisabled />
      </FormElement>
    )
  },
  {
    id: 'error',
    label: 'Error',
    element: (
      <FormElement
        labelContent="Form Element Label"
        inputId="checkbox-id-01"
        isRequired
        hasError
        errorId="checkbox-error-01"
        inlineMessage="This field is required"
      >
        <CheckboxStandalone
          id="checkbox-id-01"
          errorId="checkbox-error-01"
          isRequired
        />
      </FormElement>
    )
  },
  {
    id: 'required',
    label: 'Required',
    element: (
      <FormElement
        labelContent="Form Element Label"
        inputId="checkbox-id-01"
        isRequired
      >
        <CheckboxStandalone id="checkbox-id-01" isRequired />
      </FormElement>
    )
  },
  {
    id: 'view-mode-unchecked',
    label: 'View mode - Unchecked',
    element: (
      <FormElement labelContent="Form Element Label" isViewMode>
        <UtilityIcon
          symbol="steps"
          size="x-small"
          useCurrentColor
          assistiveText="False"
          title="False"
        />
      </FormElement>
    )
  },
  {
    id: 'view-mode-checked',
    label: 'View mode - Checked',
    element: (
      <FormElement labelContent="Form Element Label" isViewMode>
        <UtilityIcon
          symbol="check"
          size="x-small"
          useCurrentColor
          assistiveText="True"
          title="True"
        />
      </FormElement>
    )
  }
];

export let examples = [
  {
    id: 'help-text-icon',
    label: 'Help text icon',
    element: (
      <FormElement
        labelContent="Form Element Label"
        inputId="checkbox-id-01"
        hasTooltip
      >
        <CheckboxStandalone id="checkbox-id-01" />
      </FormElement>
    )
  },
  {
    id: 'required-help-text-icon',
    label: 'Required with Help text icon',
    element: (
      <FormElement
        labelContent="Form Element Label"
        inputId="checkbox-id-01"
        isRequired
        hasTooltip
      >
        <CheckboxStandalone id="checkbox-id-01" isRequired />
      </FormElement>
    )
  },
  {
    id: 'required-help-text-icon-tooltip',
    label: 'Required with Help text icon, showing tooltip',
    element: (
      <div style={{ paddingTop: '3rem' }}>
        <FormElement
          labelContent="Form Element Label"
          inputId="checkbox-id-01"
          isRequired
          hasTooltip
          showTooltip
        >
          <CheckboxStandalone id="checkbox-id-01" isRequired />
        </FormElement>
      </div>
    )
  }
];
