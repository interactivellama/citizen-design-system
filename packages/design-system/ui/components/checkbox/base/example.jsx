// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license

import React from 'react';
import classNames from 'classnames';
import _ from '../../../shared/helpers';
import { FormElement, FormElementControl } from '../../form-element';

const checkboxLabel = 'Checkbox Label';
const errorId = 'error_01';

/// ////////////////////////////////////////
// Partial(s)
/// ////////////////////////////////////////

let Demo = props => (
  <div className="demo-only" {...props}>
    {props.children}
  </div>
);

let Fieldset = props => (
  <fieldset className={classNames('slds-form-element', props.className)}>
    {props.children}
  </fieldset>
);

let Legend = props => (
  <legend className="slds-form-element__legend slds-form-element__label">
    {props.children}
  </legend>
);

export let Checkbox = props => {
  const uniqueId = _.uniqueId('checkbox-');

  return (
    <span className={classNames('slds-checkbox', props.className)}>
      {props.children}
      <input
        type="checkbox"
        name={props.name || 'options'}
        id={props.id ? props.id : uniqueId}
        value={props.id ? props.id : uniqueId}
        disabled={props.disabled}
        defaultChecked={props.checked}
        aria-describedby={props.errorId}
        tabIndex={props.tabIndex}
        aria-labelledby={
          props.labelId && props.groupId
            ? props.labelId + ' ' + props.groupId
            : null
        }
      />
      <label
        className={classNames('slds-checkbox__label')}
        htmlFor={props.id ? props.id : uniqueId}
        id={props.labelId}
      >
        <span className="slds-checkbox_faux" />
        {props.label && (
          <span
            className={classNames(
              'slds-form-element__label',
              props.hideLabel ? 'slds-assistive-text' : null
            )}
          >
            {props.label}
          </span>
        )}
      </label>
    </span>
  );
};

/// ///////////////////////////////////////////
// State Constructor(s)
/// ///////////////////////////////////////////

export let Indeterminate = props => (
  <FormElement>
    <Checkbox
      id="checkbox-indeterminate-01"
      label="Indeterminate Checkbox Label"
    />
  </FormElement>
);

export let Required = props => (
  <FormElement>
    <Checkbox label={checkboxLabel}>
      <abbr className="slds-required" title="required">
        *
      </abbr>
    </Checkbox>
  </FormElement>
);

export let ErrorState = props => (
  <FormElement
    formElementClassName="slds-has-error"
    inlineMessage="This field is required"
    errorId={errorId}
  >
    <Checkbox label={checkboxLabel} errorId="error_01">
      <abbr className="slds-required" title="required">
        *
      </abbr>
    </Checkbox>
  </FormElement>
);

export let Disabled = props => (
  <FormElement>
    <Checkbox label={checkboxLabel} disabled />
  </FormElement>
);

export let Group = props => (
  <Fieldset>
    <Legend>Checkbox Group Label</Legend>
    <FormElementControl>
      <Checkbox label={checkboxLabel} checked />
      <Checkbox label={checkboxLabel} />
    </FormElementControl>
  </Fieldset>
);

export let GroupRequired = props => (
  <Fieldset>
    <Legend>
      <abbr className="slds-required" title="required">
        *
      </abbr>{' '}
      Checkbox Group Label
    </Legend>
    <FormElementControl>
      <Checkbox label={checkboxLabel} checked />
      <Checkbox label={checkboxLabel} />
    </FormElementControl>
  </Fieldset>
);

export let GroupError = props => (
  <Fieldset className="slds-has-error">
    <Legend>
      <abbr className="slds-required" title="required">
        *
      </abbr>{' '}
      Checkbox Group Label
    </Legend>
    <FormElementControl>
      <Checkbox errorId={errorId} label={checkboxLabel} checked />
      <Checkbox errorId={errorId} label={checkboxLabel} />
    </FormElementControl>
    <div id={errorId} className="slds-form-element__help">
      This field is required
    </div>
  </Fieldset>
);

export let GroupDisabled = props => (
  <Fieldset>
    <Legend>Checkbox Group Label</Legend>
    <FormElementControl>
      <Checkbox label={checkboxLabel} disabled />
      <Checkbox label={checkboxLabel} disabled />
    </FormElementControl>
  </Fieldset>
);

/// ///////////////////////////////////////////
// Export
/// ///////////////////////////////////////////

export default (
  <FormElement>
    <Checkbox label={checkboxLabel} checked />
  </FormElement>
);

export let states = [
  {
    id: 'indeterminate',
    label: 'Indeterminate',
    element: <Indeterminate />,
    script: `
      var checkbox = document.getElementById('checkbox-indeterminate-01')
      checkbox.indeterminate = true
    `
  },
  {
    id: 'required',
    label: 'Required',
    element: <Required />
  },
  {
    id: 'error',
    label: 'Error',
    element: <ErrorState />
  },
  {
    id: 'disabled',
    label: 'Disabled',
    element: <Disabled />
  }
];

export let examples = [
  {
    id: 'group',
    label: 'Group',
    element: <Group />
  },
  {
    id: 'group-required',
    label: 'Group Required',
    element: <GroupRequired />
  },
  {
    id: 'group-error',
    label: 'Group with error',
    element: <GroupError />
  },
  {
    id: 'group-disabled',
    label: 'Group Disabled',
    element: <GroupDisabled />
  },
  {
    id: 'stacked',
    label: 'Stacked',
    element: (
      <FormElement>
        <Checkbox
          className="slds-checkbox_stacked"
          label={checkboxLabel}
          checked
        />
      </FormElement>
    )
  }
];
