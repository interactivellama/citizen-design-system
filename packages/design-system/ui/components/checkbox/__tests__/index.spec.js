// Copyright (c) 2015-present, salesforce.com, inc. All rights reserved
// Licensed under BSD 3-Clause - see LICENSE.txt or git.io/sfdc-license
/* eslint-env jest */

import React from 'react';
import createHelpers from '../../../../jest.helpers';
import { Checkbox } from '../base/example';
import { CheckboxPrimitive, CheckboxStandalone } from '../';

const { matchesMarkup } = createHelpers(__dirname);

describe('Checkbox', () => {
  it('renders checkbox', () => matchesMarkup(<Checkbox />));
  it('renders checkbox with a label', () =>
    matchesMarkup(<Checkbox label="label" />));
  it('renders checkbox with a hidden label', () =>
    matchesMarkup(<Checkbox label="label" hideLabel />));
  it('renders checkbox with required asterisk', () =>
    matchesMarkup(<Checkbox isRequired />));
  it('renders checkbox that is disabled', () =>
    matchesMarkup(<Checkbox disabled />));
  it('renders checkbox with help text icon', () =>
    matchesMarkup(<Checkbox hasTooltip />));
});

describe('Checkbox Primitive', () => {
  it('renders primitive checkbox', () => matchesMarkup(<CheckboxPrimitive />));
  it('renders primitive checkbox with an ID', () =>
    matchesMarkup(<CheckboxPrimitive id="checkbox-id" />));
  it('renders primitive checkbox with a name', () =>
    matchesMarkup(<CheckboxPrimitive name="checkbox-name" />));
  it('renders primitive checkbox with a value', () =>
    matchesMarkup(<CheckboxPrimitive value="checkbox-value" />));
  it('renders primitive checkbox that is disabled', () =>
    matchesMarkup(<CheckboxPrimitive isDisabled />));
  it('renders primitive checkbox that is required', () =>
    matchesMarkup(<CheckboxPrimitive isRequired />));
  it('renders primitive checkbox with an aria-describedby error ID', () =>
    matchesMarkup(<CheckboxPrimitive errorId="error-id" />));
});

describe('Standalone Checkbox', () => {
  it('renders standalone checkbox', () =>
    matchesMarkup(<CheckboxStandalone />));
  it('renders primitive checkbox with an ID', () =>
    matchesMarkup(<CheckboxStandalone id="checkbox-id" />));
  it('renders primitive checkbox with a name', () =>
    matchesMarkup(<CheckboxStandalone name="checkbox-name" />));
  it('renders primitive checkbox with a value', () =>
    matchesMarkup(<CheckboxStandalone value="checkbox-value" />));
  it('renders primitive checkbox that is disabled', () =>
    matchesMarkup(<CheckboxStandalone isDisabled />));
  it('renders primitive checkbox that is required', () =>
    matchesMarkup(<CheckboxStandalone isRequired />));
  it('renders primitive checkbox with an aria-describedby error ID', () =>
    matchesMarkup(<CheckboxStandalone errorId="error-id" />));
});
