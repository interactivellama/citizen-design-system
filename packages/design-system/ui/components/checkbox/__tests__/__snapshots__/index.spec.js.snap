// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Checkbox Primitive renders primitive checkbox 1`] = `
<input
  aria-describedby={undefined}
  defaultChecked={undefined}
  disabled={undefined}
  id={undefined}
  name="label"
  required={undefined}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox that is disabled 1`] = `
<input
  aria-describedby={undefined}
  defaultChecked={undefined}
  disabled={true}
  id={undefined}
  name="label"
  required={undefined}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox that is required 1`] = `
<input
  aria-describedby={undefined}
  defaultChecked={undefined}
  disabled={undefined}
  id={undefined}
  name="label"
  required={true}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with a name 1`] = `
<input
  aria-describedby={undefined}
  defaultChecked={undefined}
  disabled={undefined}
  id={undefined}
  name="checkbox-name"
  required={undefined}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with a value 1`] = `
<input
  aria-describedby={undefined}
  defaultChecked={undefined}
  disabled={undefined}
  id={undefined}
  name="label"
  required={undefined}
  type="checkbox"
  value="checkbox-value"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with an ID 1`] = `
<input
  aria-describedby={undefined}
  defaultChecked={undefined}
  disabled={undefined}
  id="checkbox-id"
  name="label"
  required={undefined}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox Primitive renders primitive checkbox with an aria-describedby error ID 1`] = `
<input
  aria-describedby="error-id"
  defaultChecked={undefined}
  disabled={undefined}
  id={undefined}
  name="label"
  required={undefined}
  type="checkbox"
  value="option"
/>
`;

exports[`Checkbox renders checkbox 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-describedby={undefined}
    aria-labelledby={null}
    defaultChecked={undefined}
    disabled={undefined}
    id="checkbox-1"
    name="options"
    tabIndex={undefined}
    type="checkbox"
    value="checkbox-1"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-1"
    id={undefined}
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
</div>
`;

exports[`Checkbox renders checkbox that is disabled 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-describedby={undefined}
    aria-labelledby={null}
    defaultChecked={undefined}
    disabled={true}
    id="checkbox-5"
    name="options"
    tabIndex={undefined}
    type="checkbox"
    value="checkbox-5"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-5"
    id={undefined}
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
</div>
`;

exports[`Checkbox renders checkbox with a hidden label 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-describedby={undefined}
    aria-labelledby={null}
    defaultChecked={undefined}
    disabled={undefined}
    id="checkbox-3"
    name="options"
    tabIndex={undefined}
    type="checkbox"
    value="checkbox-3"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-3"
    id={undefined}
  >
    <span
      className="slds-checkbox_faux"
    />
    <span
      className="slds-form-element__label slds-assistive-text"
    >
      label
    </span>
  </label>
</div>
`;

exports[`Checkbox renders checkbox with a label 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-describedby={undefined}
    aria-labelledby={null}
    defaultChecked={undefined}
    disabled={undefined}
    id="checkbox-2"
    name="options"
    tabIndex={undefined}
    type="checkbox"
    value="checkbox-2"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-2"
    id={undefined}
  >
    <span
      className="slds-checkbox_faux"
    />
    <span
      className="slds-form-element__label"
    >
      label
    </span>
  </label>
</div>
`;

exports[`Checkbox renders checkbox with help text icon 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-describedby={undefined}
    aria-labelledby={null}
    defaultChecked={undefined}
    disabled={undefined}
    id="checkbox-6"
    name="options"
    tabIndex={undefined}
    type="checkbox"
    value="checkbox-6"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-6"
    id={undefined}
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
  <div
    className="slds-form-element__icon"
  >
    <button
      aria-describedby={undefined}
      aria-haspopup={undefined}
      className="slds-button slds-button_icon"
      id={undefined}
    >
      <svg
        aria-hidden={true}
        className="slds-button__icon"
      >
        <use
          xlinkHref="/assets/icons/utility-sprite/svg/symbols.svg#info"
        />
      </svg>
      <span
        className="slds-assistive-text"
      >
        Help
      </span>
    </button>
  </div>
</div>
`;

exports[`Checkbox renders checkbox with required asterisk 1`] = `
<div
  className="slds-checkbox"
>
  <input
    aria-describedby={undefined}
    aria-labelledby={null}
    defaultChecked={undefined}
    disabled={undefined}
    id="checkbox-4"
    name="options"
    tabIndex={undefined}
    type="checkbox"
    value="checkbox-4"
  />
  <label
    className="slds-checkbox__label"
    htmlFor="checkbox-4"
    id={undefined}
  >
    <span
      className="slds-checkbox_faux"
    />
  </label>
</div>
`;

exports[`Standalone Checkbox renders primitive checkbox that is disabled 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby={undefined}
    defaultChecked={undefined}
    disabled={true}
    id={undefined}
    name="label"
    required={undefined}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox that is required 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby={undefined}
    defaultChecked={undefined}
    disabled={undefined}
    id={undefined}
    name="label"
    required={true}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with a name 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby={undefined}
    defaultChecked={undefined}
    disabled={undefined}
    id={undefined}
    name="checkbox-name"
    required={undefined}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with a value 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby={undefined}
    defaultChecked={undefined}
    disabled={undefined}
    id={undefined}
    name="label"
    required={undefined}
    type="checkbox"
    value="checkbox-value"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with an ID 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby={undefined}
    defaultChecked={undefined}
    disabled={undefined}
    id="checkbox-id"
    name="label"
    required={undefined}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders primitive checkbox with an aria-describedby error ID 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby="error-id"
    defaultChecked={undefined}
    disabled={undefined}
    id={undefined}
    name="label"
    required={undefined}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;

exports[`Standalone Checkbox renders standalone checkbox 1`] = `
<span
  className="slds-checkbox slds-checkbox_standalone"
>
  <input
    aria-describedby={undefined}
    defaultChecked={undefined}
    disabled={undefined}
    id={undefined}
    name="label"
    required={undefined}
    type="checkbox"
    value="option"
  />
  <span
    className="slds-checkbox_faux"
  />
</span>
`;
